smdebug/exceptions.py

```diff
@@ -2,6 +2,14 @@
 from smdebug.core.modes import ModeKeys as modes
 
 
+class SMDebugError(Exception):
+    pass
+
+
+class SMDebugCustomerError(SMDebugError):
```

Comment:
when will we have a customer error?

Reply:
this will be reserved for things like incorrect hook configuration (e.g. missing hook initialization on training running on EC2 rather than sagemaker, since sagemaker automatically creates a JSON config but EC2 won't).
For now, we'll classify all errors as SMDebugError and fill this in as necessary.

Reply:
LGTM